clang-tools-extra/test/clang-tidy/checkers/modernize/use-emplace.cpp

```diff
@@ -1,4 +1,4 @@
-// RUN: %check_clang_tidy %s modernize-use-emplace %t -- \
+// RUN: %check_clang_tidy %s -std=c++17 modernize-use-emplace %t -- \
```

Comment:
check has `LangOpts.CPlusPlus11`, so this need to be removed
```suggestion
// RUN: %check_clang_tidy %s modernize-use-emplace %t -- \
```

Reply:
Yeah, this is one of the thing I wasn't sure at all, let me explain:

Running this check on version earlier than C++20, results in cases like this (specified on the regression file):

```c++
  // These should not be noticed or fixed; after the correction, the code won't
  // compile.
  v1.push_back(NonTrivialNoCtor{""});
  v1.push_back({""});
  v1.push_back(NonTrivialNoCtor{InnerType{""}});
  v1.push_back({InnerType{""}});
  v1.emplace_back(NonTrivialNoCtor{""});

```

AFAIK If we remove `-std=c++17` the tests will run on every standard other than c++20 where these cases would fail(the diagnosting would be deem invalid). Would it make sense to explicitly restrict it to a specific version instead? or is there any other alternative?

Reply:
On second thought would this be sufficient for the cases shown above?:
I've noticed that `use-ranges.cpp` uses  a custom suffix bound to a standard
```cpp
// RUN: %check_clang_tidy -std=c++23 %s modernize-use-ranges %t -check-suffixes=,CPP23 -- -I %S/Inputs/
// CHECK-FIXES-CPP23: #include <numeric>

```
I could implement cases with designated initializer in a separate  c++20 file.

Reply:
I'd suggest to use `-std=c++11` in "base" RUN command - that is the lowest standard we need to support.
Then add 2nd RUN command with `-std=c+20` and `-check-suffixes=,CPP20` - that will make sure that previous test cases are not broken due to upcoming changes with `c++20` and some new test-cases are matched with custom-suffix (that would be `v1.push_back` from your prev comment).

If you need something from `c++20` standard that is not supported by compiler in prevous standarts, I'd suggest to use `#if __cplusplus >= 202002L` as in `concat-nested-namespaces.cpp` or create a separate file. But I guess that only designated initializers are needed from `c++20` so a small `#if` should be used.

Reply:
Thx for your suggestions, this is really great, this should be addressed now.